ui/pages/confirmations/components/confirm/info/typed-sign/permit-simulation/permit-simulation.tsx

```diff
@@ -69,44 +45,31 @@ const PermitSimulation: React.FC<{
       description={t('permitSimulationDetailInfo')}
       simulationHeading={t('spendingCap')}
       simulationElements={
-        <>
-          <Box display={Display.Flex}>
-            <Box
-              display={Display.Inline}
-              marginInlineEnd={1}
-              minWidth={BlockSize.Zero}
-            >
-              <Tooltip
-                position="bottom"
-                title={tokenValueMaxPrecision}
-                wrapperStyle={{ minWidth: 0 }}
-                interactive
-              >
-                <Text
-                  data-testid="simulation-token-value"
-                  backgroundColor={BackgroundColor.backgroundAlternative}
-                  borderRadius={BorderRadius.XL}
-                  paddingInline={2}
-                  style={{ paddingTop: '1px', paddingBottom: '1px' }}
-                  textAlign={TextAlign.Center}
-                >
-                  {shortenString(tokenValue || '', {
-                    truncatedCharLimit: 15,
-                    truncatedStartChars: 15,
-                    truncatedEndChars: 0,
-                    skipCharacterInEnd: true,
-                  })}
-                </Text>
-              </Tooltip>
-            </Box>
-            <Name value={verifyingContract} type={NameType.ETHEREUM_ADDRESS} />
-          </Box>
-          <Box>
-            {fiatValue && (
-              <IndividualFiatDisplay fiatAmount={fiatValue} shorten />
+        tokenDetails ? (
+          <Box
+            display={Display.Flex}
+            flexDirection={FlexDirection.Column}
+            gap={2}
+          >
+            {tokenDetails.map(
+              (
+                { token, amount }: { token: string; amount: string },
+                i: number,
+              ) => (
+                <PermitSimulationValueDisplay
+                  key={`${token}-${i}`}
+                  tokenContract={token}
+                  value={amount}
+                />
```

Comment:
Will `tokenDetails.map` work for even single token ?

Reply:
I think I did this right unless I missed something. If it is a PermitSingle, I set it in an array for tokenDetail 
```
  if (isPermitSingle) {
    tokenDetails = [details];
  } else if (isPermitBatch) {
    tokenDetails = details;
  }
```